server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -775,9 +780,13 @@ private static ActionListener<SearchResponse> createCCSListener(
         ) {
             @Override
             void innerOnResponse(SearchResponse searchResponse) {
-                // TODO: in CCS fail fast ticket we may need to fail the query if the cluster gets marked as FAILED
                 ccsClusterInfoUpdate(searchResponse, clusters, clusterAlias, skipUnavailable);
                 searchResponseMerger.add(searchResponse);
+                if (progressListener != null && progressListener != SearchProgressListener.NOOP) {
```

Comment:
can you remind me when the progress listener is actually null?

Reply:
I think there were tests that were doing it, but I have the null check and cleaned up the tests. So far it looks good so I have that in the next push.

Reply:
great!